tests/system/src/test/java/test/com/sun/javafx/animation/SynchronisityTest.java

```diff
@@ -0,0 +1,90 @@
+package test.com.sun.javafx.animation;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.stage.Stage;
+import test.util.Util;
+
+// Based on https://bugs.openjdk.org/browse/JDK-8159048
+public class SynchronisityTest extends Application {
+
+    public static void main(String[] args) {
+        Application.launch(args);
+    }
+
+    private static final CountDownLatch startupLatch = new CountDownLatch(1);
+    private static Stage primaryStage;
+
+    @Override
+    public void start(Stage stage) throws Exception {
+        primaryStage = stage;
+        startupLatch.countDown();
+    }
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        Util.launch(startupLatch, SynchronisityTest.class);
+    }
+
+    @AfterAll
+    public static void shutdown() {
+        Util.shutdown(primaryStage);
+    }
+
+    /**
+     * Number of seconds to wait for a failure. If an exception is not thrown in this time, it's assumed it won't be
+     * thrown later too.
+     */
+    private static final int GRACE_PERIOD = 15;
+
+    private AtomicBoolean failed = new AtomicBoolean(false);
+    private CountDownLatch waiter = new CountDownLatch(1);
+    private ExecutorService executor = Executors.newCachedThreadPool();
+
+    private Thread thread;
+    private Throwable throwable;
+
+    protected void runTest(Runnable runnable) throws InterruptedException {
+        Platform.runLater(() -> registerExceptionHandler());
+
+        for (int i = 0; i < 10; i++) {
+            executor.submit(runnable);
+        }
+
+        waiter.await(GRACE_PERIOD, TimeUnit.SECONDS);
```

Comment:
I don't think this will do what you want. This will return almost immediately, before any of the animation has run (as soon as the runnable that sets the uncaught exception handler finished). What I think you need is something more like this:

```
        Platform.runLater(() -> registerExceptionHandler());
        assertTrue(waiter.await(GRACE_PERIOD, TimeUnit.SECONDS));

        // start the 10 threads...

        Util.sleep(GRACE_PERIOD);
```

The await ensures that the uncaught exception handler has been registered before starting the threads. The sleep then runs the threads for a fixed amount of time.

Reply:
> I don't think this will do what you want. This will return almost immediately, before any of the animation has run (as soon as the runnable that sets the uncaught exception handler finished).

I observed that this setup works correctly for the `AnimationTimer` test. The test does wait for `GRACE_PERIOD` seconds when there is no exception and then succeeds, or catches and fails immediately when there onw is thrown. The `Animation` test was more finicky, but I think that is because I'm not simulating a processing load properly.

> The await ensures that the uncaught exception handler has been registered before starting the threads. 

While technically true, the handler register in a negligible amount of time compared to the grace period. Because exceptions are thrown continuously )it's not a one-time event), the handler will never miss a failure.

I will change it for correctness sake.